src/matrices.jl

```diff
@@ -282,7 +282,7 @@ See also: [`CrossRecurrenceMatrix`](@ref).
 """
 function JointRecurrenceMatrix(x, y, ε; kwargs...)
     n = min(size(x,1), size(y,1))
-    rm1 = RecurrenceMatrix( (@view x[1:n,:]), ε, kwargs...)
-    rm2 = RecurrenceMatrix( (@view y[1:n,:]), ε, kwargs...)
+    rm1 = RecurrenceMatrix( (x[1:n,:]), ε, kwargs...)
```

Comment:
Here we should add an if clause:
```
if n == size(x, 1) && n == size(y, 1)
 RecurrenceMatrix(x, ...)
 ...
else
 # the code that exists now
end
```
to avoid making a copy in the most common case.

Reply:
Not exactly: the old code works if `x` or `y` is `AbstractArray`, but not if it is a `Dataset`.
So, better make two different functions - depending on the type.